apps/condo/domains/property/components/panels/Builder/BuildingPanelEdit.tsx

```diff
@@ -960,16 +960,21 @@ const UnitForm: React.FC<IPropertyMapModalForm> = ({ builder, refresh }) => {
             isUnitLabelUnique = builder.validateUniqueUnitLabel()
         } else if (mode === 'editUnit') {
             const selectedUnit = builder.getSelectedUnit()
-            const unitPlacementChanged = selectedUnit.floor !== floor || selectedUnit.section !== section
-            const labelValidation = selectedUnit.label !== label
+            const unitPlacementChanged = selectedUnit.floor !== floor
+                || selectedUnit.section !== section
+                || unitType !== selectedUnit.unitType
+            const labelChanged = selectedUnit.label !== label
+            const labelValidation = labelChanged
                 ? builder.validateUniqueUnitLabel(label, 'section')
-                : true
+                : false
 
-            isUnitLabelUnique = unitPlacementChanged && labelValidation
+            isUnitLabelUnique = unitPlacementChanged
+                ? unitPlacementChanged && (!labelChanged || labelValidation )
+                : labelValidation
```

Comment:
Why use `unitPlacementChanged` twice?

Reply:
@Alllex202 It results from the unit uniqueness validation logic. 
So we have already created map and we want to edit unit. First of all we should check may any fields of unit has been changed. There it is - unitType, section or floor. Second part is check that label of the unit is unique by all of the sections.
Otherwise we only check label uniqueness of the edited unit